src/compute/src/sink/copy_to_s3_oneshot.rs

```diff
@@ -67,68 +81,266 @@ where
 }
 
 fn copy_to<G>(
-    sinked_collection: Collection<G, Row, Diff>,
+    input_collection: Collection<G, Row, Diff>,
     err_collection: Collection<G, DataflowError, Diff>,
     sink_id: GlobalId,
     up_to: Antichain<G::Timestamp>,
+    sink_connection: CopyToS3OneshotSinkConnection,
+    connection_context: ConnectionContext,
     response_protocol_handle: Rc<RefCell<Option<ResponseProtocol>>>,
 ) where
     G: Scope<Timestamp = Timestamp>,
 {
-    let name = format!("s3-oneshot-{}", sink_id);
-    let mut op = OperatorBuilder::new(name, sinked_collection.scope());
-    let mut ok_input = op.new_input(&sinked_collection.inner, Pipeline);
-    let mut err_input = op.new_input(&err_collection.inner, Pipeline);
-
-    op.build(|_cap| {
-        let mut row_count: u64 = 0;
-        let mut finished = false;
-        let mut ok_buf = Default::default();
-
-        move |frontiers| {
-            if finished {
-                // Drain the inputs, to avoid the operator being constantly rescheduled
-                ok_input.for_each(|_, _| {});
-                err_input.for_each(|_, _| {});
-                return;
+    let scope = input_collection.scope();
+    let worker_id = scope.index();
+
+    let mut builder = AsyncOperatorBuilder::new("CopyToS3".to_string(), input_collection.scope());
```

Comment:
OOC would it show up in coverage if it runs successfully?

Reply:
@antiguru This can't be written in any other way but also this isn't meant to be maintained compute. What exactly is the problem with the async operator?

Reply:
The problem is that while it's in the async call to upload the artifact, it'll not consume its inputs and progress information. This can cause unpredictable behavior because some data can linger longer in memory than expected.

I think an alternative design is to move the async tasks off the timely thread and use queues to communicate updates in between.

Reply:
> this isn't meant to be maintained compute

I'd like to propose a solution that has the potential to make everyone not unhappy: How above we move this operator into `mz-storage-operators`? That way the code ownership is clear and we don't risk ending up with a situation like `persist_sink` that nobody feels responsible for.

It makes sense to me to have this operator maintained by the Storage/S&S team given that it interfaces with an external system.

Reply:
> How above we move this operator into mz-storage-operators?

That sounds good to me! I will move it there in the meantime and the discussion can happen in parallel.